### DISCUSSION TITLE
Feature Request: JSON extension recurse into nested JSON for functions such as `json_keys`, as it already does with `json_structure`

### DISCUSSION THREAD

**Discussion by LeeBrotherston** [duckdb/duckdb#13882] [Ideas]
Currently using `json_keys` like so:

```sql
SELECT json_keys('{ "color": "blue", "sound": "moo", "nested": { "hello": "dave", "super": "duper" } }');
```

Yields the result:

```
[color, sound, nested]
```

Whereas `json_structure` will recurse into the nested JSON and produce....

```
{"color":"VARCHAR","sound":"VARCHAR","nested":{"hello":"VARCHAR","super":"VARCHAR"}}
```

It would be really helpful to be able to obtain all keys from the JSON structure when using `json_keys` in a similar fashion (be that via a new function, or modification to the existing).

**Comment by cmdlineluser** [comment]
SQLite's `json_tree()` may serve as an existing example of similar functionality.

- https://github.com/duckdb/duckdb/discussions/11194
